python-sdk/src/astro/databases/aws/redshift.py

```diff
@@ -264,4 +283,89 @@ def is_native_load_file_available(
         """
-        return False
+        file_type = NATIVE_PATHS_SUPPORTED_FILE_TYPES.get(source_file.type.name)
+        location_type = self.NATIVE_PATHS.get(source_file.location.location_type)
+        return bool(location_type and file_type)
+
+    def load_file_to_table_natively(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        native_support_kwargs: Optional[Dict] = None,
+        **kwargs,
+    ):
+        """
+        Checks if optimised path for transfer between File location to database exists
+        and if it does, it transfers it and returns true else false.
+
+        :param source_file: File from which we need to transfer data
+        :param target_table: Table that needs to be populated with file data
+        :param if_exists: Overwrite file if exists. Default False
+        :param native_support_kwargs: kwargs to be used by method involved in native support flow
+        """
+        method_name = self.NATIVE_PATHS.get(source_file.location.location_type)
+        if method_name:
+            transfer_method = self.__getattribute__(method_name)
+            transfer_method(
+                source_file=source_file,
+                target_table=target_table,
+                if_exists=if_exists,
+                native_support_kwargs=native_support_kwargs,
+                **kwargs,
+            )
+        else:
+            raise DatabaseCustomError(
+                f"No transfer performed since there is no optimised path "
+                f"for {source_file.location.location_type} to bigquery."
+            )
+
+    def load_s3_file_to_table(
+        self,
+        source_file: File,
+        target_table: Table,
+        native_support_kwargs: Optional[Dict] = None,
+        **kwargs,
+    ):
+        """
+        Load content of multiple files in S3 to output_table in Redshift by:
+        - Creating a table
+        - Using the COPY command
+
+        :param source_file: Source file that is used as source of data
+        :param target_table: Table that will be created on the redshift
+        :param if_exists: Overwrite table if exists. Default 'replace'
+        :param native_support_kwargs: kwargs to be used by method involved in native support flow
+
+        .. seealso::
+            `Redshift official documentation on CREATE TABLE
+            <https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_NEW.html>`_
+            `Redshift official documentation on COPY
+            <https://docs.aws.amazon.com/redshift/latest/dg/r_COPY.html>`_
+        """
+        native_support_kwargs = native_support_kwargs or {}
+
+        table_name = self.get_table_qualified_name(target_table)
+        file_type = NATIVE_PATHS_SUPPORTED_FILE_TYPES.get(source_file.type.name)
+
+        iam_role = native_support_kwargs.pop("IAM_ROLE", None)
+        if not iam_role:
+            raise TypeError(
```

Comment:
```suggestion
            raise ValueError(
```

Reply:
@pankajkoti I think this should be ValueError since the value is expected as input and is not passed.

Reply:
I believe this should be TypeError. I too was confused on whether it should be ValueError or TypeError, but value error is when an invalid value is passed. Here, Python gives a Typ
<img width="723" alt="Screenshot 2022-09-02 at 12 12 39 PM" src="https://user-images.githubusercontent.com/10206082/188074902-54d3307a-a39c-4e68-b3c5-6b28b96208da.png">
eError when an expected argument itself is not passed.